src/components/SortElement/SortElement.js

```diff
@@ -1,25 +1,59 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 
+import { connect } from 'react-redux';
 import styles from './SortElement.module.scss';
 import { SORT_BY_VALUES } from '~/services/mock-data';
+import { getMovies } from '../../store/actions/movies-actions';
+import { UI_SET_SORTING } from '../../store/actions/search-actions';
 
-const SortElement = () => (
-  <div className={styles.SortElement}>
-    <p>Sort By</p>
-    <p>
-      <select className={styles.dropdown}>
-        {SORT_BY_VALUES.map((opt) => (
-          <option key={opt.id} value={opt.value}>
-            {opt.label}
-          </option>
-        ))}
-      </select>
-    </p>
-  </div>
-);
+const SortElement = ({
+  sortBy: currentSortBy, sortOrder: currentSortOrder, fetchMovies, setSorting,
+}) => {
+  const [sortBy, setSortBy] = useState(currentSortBy);
+  const [sortOrder, setSortOrder] = useState(currentSortOrder);
+
+  const handleChange = (event) => {
+    const [newSortBy, newSortOrder] = event.target.value.split(',');
+    setSortBy(newSortBy);
+    setSortOrder(newSortOrder);
+    setSorting(newSortBy, newSortOrder);
+    fetchMovies();
+  };
+
+  const selectedValue = useMemo(() => `${sortBy},${sortOrder}`, [sortBy, sortOrder]);
```

Comment:
I'm just not sure that this `useMemo` make sense

Reply:
Indeed. Thanks, fixed